main.tf

```diff
@@ -1,3 +1,17 @@
+locals {
+  fqdn                    = var.hostname != "" ? format("%s.%s", var.hostname, var.domain) : var.domain
+  san                     = formatlist("%s.%s", var.subject_alternative_names, var.domain)
+  skip_route53_validation = var.validation_method == "EMAIL" ? true : var.skip_route53_validation
+  create_route53_record   = var.create_route53_record && false == local.skip_route53_validation
+}
+
+data "aws_route53_zone" "default" {
+  for_each = toset((local.create_route53_record) ? [var.domain] : [])
```

Comment:
I don't understand the need to change this from a `count` to a `for_each`. What is the objective here?